File: zad2_3/app/src/main/java/com/pam/bmi/ui/graph/GraphFragment.java
package com.pam.bmi.ui.graph;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.pam.bmi.R;

import androidx.fragment.app.Fragment;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;

import java.util.ArrayList;

public class GraphFragment extends Fragment {

    LineChart lineChart;
    LineData lineData;
    LineDataSet lineDataSet;
    ArrayList<Entry> lineEntries;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_graph, container, false);

        lineChart = view.findViewById(R.id.lineChart);
        lineChart.getXAxis().setPosition(XAxis.XAxisPosition.BOTTOM);
        lineChart.getAxisLeft().setDrawGridLines(false);
        lineChart.getAxisRight().setDrawGridLines(false);
        lineChart.getDescription().setText("BMI/Months");

        getEntries();

        lineDataSet = new LineDataSet(lineEntries, "history bmi values");
        lineDataSet.setColor(Color.GREEN);
        lineDataSet.setLineWidth(2f);
        lineDataSet.setValueTextColor(Color.BLACK);

        lineData = new LineData(lineDataSet);
        lineChart.setData(lineData);
        lineChart.invalidate();
        lineDataSet.setValueTextColor(Color.BLACK);
        lineDataSet.setValueTextSize(17f);

        return view;
    }

    private void getEntries() {
        lineEntries = new ArrayList<>();
        lineEntries.add(new Entry(1, 16.6f));
        lineEntries.add(new Entry(2, 15.7f));
        lineEntries.add(new Entry(3, 16.5f));
        lineEntries.add(new Entry(4, 16.9f));
        lineEntries.add(new Entry(5, 17.5f));
        lineEntries.add(new Entry(6, 18.5f));
        lineEntries.add(new Entry(7, 19.1f));
        lineEntries.add(new Entry(8, 20.7f));
        lineEntries.add(new Entry(9, 21.3f));
        lineEntries.add(new Entry(10, 18.5f));
        lineEntries.add(new Entry(11, 19.4f));
        lineEntries.add(new Entry(12, 17.1f));
    }
}

File: zad_4_kotlin/app/src/main/java/com/pam/bmi/ui/graph/GraphFragment.kt
package com.pam.bmi.ui.graph

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.data.LineData
import com.github.mikephil.charting.data.LineDataSet
import com.pam.bmi.R

class GraphFragment : Fragment() {
    private var lineChart: LineChart? = null
    private var lineData: LineData? = null
    private var lineDataSet: LineDataSet? = null
    private var lineEntries: ArrayList<Entry>? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_graph, container, false)
        lineChart = view.findViewById(R.id.lineChart)
        lineChart?.xAxis?.position = XAxis.XAxisPosition.BOTTOM
        lineChart?.axisLeft?.setDrawGridLines(false)
        lineChart?.axisRight?.setDrawGridLines(false)
        lineChart?.description?.text = "BMI/Months"
        generateEntries()
        lineDataSet = LineDataSet(lineEntries, "history bmi values")
        lineDataSet?.color = Color.GREEN
        lineDataSet?.lineWidth = 2f
        lineDataSet?.valueTextColor = Color.BLACK
        lineData = LineData(lineDataSet)
        lineChart?.data = lineData
        lineChart?.invalidate()
        lineDataSet?.valueTextColor = Color.BLACK
        lineDataSet?.valueTextSize = 17f
        return view
    }

    private fun generateEntries() {
        lineEntries = ArrayList()
        lineEntries?.apply {
            add(Entry(1f, 16.6f))
            add(Entry(2f, 15.7f))
            add(Entry(3f, 16.5f))
            add(Entry(4f, 16.9f))
            add(Entry(5f, 17.5f))
            add(Entry(6f, 18.5f))
            add(Entry(7f, 19.1f))
            add(Entry(8f, 20.7f))
            add(Entry(9f, 21.3f))
            add(Entry(10f, 18.5f))
            add(Entry(11f, 19.4f))
            add(Entry(12f, 17.1f))
        }
    }
}
